packages/evm/contracts/budgets/ManagedBudgetWithFees.sol

```diff
@@ -0,0 +1,266 @@
+    function disburse(bytes calldata data_)
+        public
+        virtual
+        override(ABudget, ManagedBudget)
+        onlyAuthorized
+        returns (bool)
+    {
+        Transfer memory request = abi.decode(data_, (Transfer));
+        if (request.assetType == AssetType.ERC20 || request.assetType == AssetType.ETH) {
+            FungiblePayload memory payload = abi.decode(request.data, (FungiblePayload));
+
+            uint256 maxManagementFee = payload.amount * managementFee / 10_000;
+
+            uint256 avail = available(request.asset);
+            if ((payload.amount + maxManagementFee) > avail) {
+                revert InsufficientFunds(request.asset, avail, payload.amount + maxManagementFee);
+            }
+
+            incentiveFees[request.target] = maxManagementFee;
+            reservedFunds[request.asset] += maxManagementFee;
+            _distributedFungible[request.asset] += payload.amount;
+            _transferFungible(request.asset, request.target, payload.amount);
+        } else if (request.assetType == AssetType.ERC1155) {
+            ERC1155Payload memory payload = abi.decode(request.data, (ERC1155Payload));
+
+            uint256 avail = IERC1155(request.asset).balanceOf(address(this), payload.tokenId);
+            if (payload.amount > avail) {
+                revert InsufficientFunds(request.asset, avail, payload.amount);
+            }
+
+            _transferERC1155(request.asset, request.target, payload.tokenId, payload.amount, payload.data);
+        } else {
+            return false;
+        }
+
+        return true;
+    }
+
+    /// @notice Pays the management fee for a specific boost and incentive
+    /// @param boostId The ID of the boost for which the management fee is being paid
+    /// @param incentiveId The ID of the incentive within the boost
+    /// @dev The function checks the type of incentive and ensures that the claims have reached the limit
+    /// or the balance is zero before transferring the management fee to the boost owner
+    /// @dev Supports management fee payouts for AERC20Incentive and AERC20VariableIncentive deployments
+    function payManagementFee(uint256 boostId, uint8 incentiveId) external {
+        BoostLib.Boost memory boost = core.getBoost(boostId);
+
+        address validIncentive = address(boost.incentives[incentiveId]);
+
+        if (AIncentive(validIncentive).supportsInterface(type(AERC20Incentive).interfaceId)) {
+            uint256 claims = AERC20Incentive(validIncentive).claims();
+            uint256 limit = AERC20Incentive(validIncentive).limit();
+            uint256 balanceRemaining = AIncentive(validIncentive).asset().balanceOf(validIncentive);
+            if (claims == limit || balanceRemaining == 0) {
+                _transferManagementFee(boostId, incentiveId, boost, claims, limit);
+                return;
+            }
+            revert BoostError.Unauthorized();
```

Comment:
Would it be worth it to add a more descriptive error for when the incentive still has claims available?